### PR TITLE
feat: add call on extension

### PR DESCRIPTION
**Pull request by KedziaPawel**
## PR-Codex overview
This PR adds new functions and types related to integrations, updates enums and constants, and adds new tests. 

### Detailed summary
- Adds `createTestRevert` function to `error.ts`.
- Adds `LEND_SELECTOR` and `REDEEM_SELECTOR` constants to `selectors.ts`.
- Adds `IntegrationManagerActionId` and `Integration` enums to `enums.ts`.
- Adds `CallArgsForIntegration` type and `encodeCallArgsForIntegration` and `decodeCallArgsForIntegration` functions to `callArgs.ts`.
- Adds `IntegrationDataForAaveV2Lend`, `CallArgsForAaveV2Lend`, `AaveV2LendTrade`, `encodeIntegrationDataForAaveV2Lend`, `decodeIntegrationDataForAaveV2Lend`, `encodeCallArgsForAaveV2Lend`, `decodeCallArgsForAaveV2Lend`, `prepareCallOnAaveV2LendParams`, `IntegrationDataForAaveV2Redeem`, `CallArgsForAaveV2Redeem`, `AaveV2RedeemTrade`, `encodeIntegrationDataForAaveV2Redeem`, `decodeIntegrationDataForAaveV2Redeem`, `encodeCallArgsForAaveV2Redeem`, `decodeCallArgsForAaveV2Redeem`, and `prepareCallOnAaveV2RedeemParams` functions to `aaveV2.ts`.
- Adds `PrepareTradeParams` type and `prepareAdapterTrade` function to `prepareAdapterTrade.ts`.
- Updates `VaultAction`, `RateAsset`, `FeeType`, and `FeeAmount` enums in `enums.ts`.
- Adds `FEE_MANAGER`, `INTEGRATION_MANAGER`, and `AAVE_V2_ADAPTER` constants to `constants.ts`.
- Adds new tests to various files.

> The following files were skipped due to too many changes: `packages/sdk/src/actions/claimOwnership.test.ts`, `packages/sdk/src/actions/buyShares.test.ts`, `packages/sdk/src/actions/redeemSharesInKind.test.ts`, `packages/sdk/src/actions/prepareAdapterTrade.test.ts`, `packages/sdk/src/integrations/callArgs.test.ts`, `packages/sdk/src/actions/redeemSharesForSpecificAssets.test.ts`, `packages/sdk/src/integrations/aaveV2.ts`, `packages/sdk/src/errors/getErrorCode.ts`, `packages/sdk/src/integrations/aaveV2.test.ts`, `packages/sdk/reverts.txt`, `packages/sdk/src/errors/errors.test.ts`, `packages/sdk/src/errors/errorCodes.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/sdk/src/actions/callOnExtension.test.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  const integrationData = encodeAbiParameters(parseAbiParameters("address aToken, uint depositAmount"), [
+    A_WETH,
+    depositAmount,
+  ]);
+
+  const callArgs = encodeAbiParameters(parseAbiParameters("address adapter, bytes4 selector, bytes integrationData"), [
+    AAVE_V2_ADAPTER,
+    lendSelector,
+    integrationData,
+  ]);
```
Comment (fubhy) [inline packages/sdk/src/actions/callOnExtension.test.ts:39]: Can we also get utils for this? So end users wouldn't have to write this themselves?

File: packages/sdk/tests/constants.ts [hunk 2]
```diff
@@ -1,4 +1,8 @@
 export const DEPLOYER = "0x4f1C53F096533C04d8157EFB6Bca3eb22ddC6360";
+export const FEE_MANAGER = "0xAf0DFFAC1CE85c3fCe4c2BF50073251F615EefC4";
+export const INTEGRATION_MANAGER = "0x31329024f1a3E4a4B3336E0b1DfA74CC3FEc633e";
+export const AAVE_V2_ADAPTER = "0xECe6B376af7C9273cebaf6528565c47Ea2Cb8a4C";
+export const A_WETH = "0x030bA81f1c18d280636F32af80b9AAd02Cf0854e"; //
```
Comment (fubhy) [inline packages/sdk/tests/constants.ts:5]: What's that comment?

**Review by fubhy — CHANGES_REQUESTED** [review]

File: packages/sdk/src/integrations/aaveV2.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { LEND_SELECTOR } from "../constants/selectors.js";
+
+const integrationDataForAaveV2LendAbiParamaters = parseAbiParameters("address aToken, uint depositAmount");
+
+export interface IntegrationDataForAaveV2Lend {
+  aToken: Address;
+  depositAmount: bigint;
+}
+
+export function encodeIntegrationDataForAaveV2Lend({ aToken, depositAmount }: IntegrationDataForAaveV2Lend) {
```
Comment (rayphelan) [inline packages/sdk/src/integrations/aaveV2.ts:13]: You could add the `Hex` return type to this function

File: packages/sdk/src/errors/errorCodes.ts [hunk 4]
```diff
@@ -204,4 +262,149 @@ export const errorDictionary: {
...
+  },
+  [ENABLING_POLICY_RESTRICTS_ACTIONS_OF_CURRENT_INVESTORS]: {
+    code: ENABLING_POLICY_RESTRICTS_ACTIONS_OF_CURRENT_INVESTORS,
+    label: "Policy restricts actions of current investors",
+    description: "Policy restricts actions of current investors",
+  },
+  [POLICIES_AND_SETTINGS_DATA_ARRAY_ARE_UNEQUAL]: {
+    code: POLICIES_AND_SETTINGS_DATA_ARRAY_ARE_UNEQUAL,
+    label: "Policies and settingsData array lengths unequal",
+    description: "Policies and settingsData array lengths unequal",
```
[the remark this answers is not in the document]
Comment (fubhy) [reply packages/sdk/src/errors/errorCodes.ts:398]: We can also accept it as this for now and revisit all these description in one go at some point with a native speaker?

File: packages/sdk/src/constants/selectors.ts [hunk 5]
```diff
@@ -12,3 +14,5 @@ export const PRICELESS_ASSET_BYPASS_START_ASSET_BYPASS_TIMELOCK_SELECTOR = "0x07
 export const SYNTHETIX_ASSIGN_EXCHANGE_DELEGATE_SELECTOR = "0x447fbc63"; // approveExchangeOnBehalf(address)
 export const SETTLE_CONTINUOUS_FEES_SELECTOR = "0xc071fe93"; // settleContinuousFees(address,bytes)
 export const VAULT_CALL_ANY_DATA_HASH = "0x5bf1898dd28c4d29f33c4c1bb9b8a7e2f6322847d70be63e8f89de024d08a669"; // mln.vaultCall.any
+export const LEND_SELECTOR = getFunctionSelector("lend(address,bytes,bytes)");
+export const REDEEM_SELECTOR = getFunctionSelector("redeem(address,bytes,bytes)");
```
Comment (fubhy) [inline packages/sdk/src/constants/selectors.ts:18]: Let's avoid function calls in `constants.ts`. Please copy& paste the result of the function call here. And add a comment afterwards.

File: packages/sdk/tests/error.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+      abi: [],
+      functionName: "",
+      message,
+    }),
+    {
+      abi: [],
+      functionName: "",
+    },
+  );
+}
```
Comment (fubhy) [inline packages/sdk/tests/error.ts:15]: `createTestRevert`

File: packages/sdk/src/errors/errors.test.ts [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+  expect(catchError(createMockedError("validatePolicies: Caller not allowed"))).toEqual(
+    new EnzymeError(POLICY_RULE_INAVLID_CALLER_NOT_ALLOWED_TO_PERFORM_CALL),
+  );
+});
+
+test("should parse properly catched error: __enablePolicyForFund: Policy is already enabled", () => {
+  expect(catchError(createMockedError("__enablePolicyForFund: Policy is already enabled"))).toEqual(
+    new EnzymeError(POLICY_ALREADY_ENABLED),
+  );
+});
```
Comment (fubhy) [inline packages/sdk/src/errors/errors.test.ts:208]: You can use `test.each` for this.

**Review by fubhy — CHANGES_REQUESTED** [review]

**Pull request merged**